EIPS/eip-5089.md

```diff
@@ -0,0 +1,340 @@
+  Has units defined by the corresponding ERC-20 contract.
+- maturity: The timestamp (unix) at which a Principal Token matures. Principal Tokens become redeemable for underlying at or after this timestamp.
+- exchangeRate: A variable used to modulate the redeemable ratio of underlying <-> Principal Token.
+- fee: An amount of underlying or Principal Token charged to the user by the Principal Token. Fees can exists on redemption or post-maturity yield. _Alberto: I'm not sure we need this in the ERC._
+
+### Methods
+
+#### underlying
+
+The address of the underlying token used by the Principal Token for accounting, depositing, and withdrawing.
+
+MUST be an ERC-20 token contract.
+
+MUST _NOT_ revert.
+
+```yaml
+- name: underlying
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: underlyingAddress
+      type: address
+```
+
+#### maturity
+
+The unix timestamp (uint256) after which Principal Tokens can be redeemed for their underlying deposit according to `exchangeRate`.
+
+MUST _NOT_ revert.
+
+```yaml
+- name: maturity
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: timestamp
+      type: uint256
+```
+
+#### maturityRate
+
+The exchangeRate during the block that a first user first calls `redeem` or `mature` after maturity.
+
+OPTIONAL - This method can be used to provide needed data for the calculation of interest post-maturity, but interfaces and other contracts MUST NOT expect this method to be present. _Alberto: Optionality in ERCs is better avoided. Not everyone will implement it, therefore integrators must assume it might not be there, and leads to a proliferation of `safe*` methods._
+
+MUST _NOT_ revert.
+
+```yaml
+- name: maturityRate
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: rate
+      type: uint256
+```
+
+#### exchangeRate
+
+The current exchangeRate of the underlying token <-> money market deposit post-maturity. Returns 0 before maturity.
+_Alberto: I wonder if we could make the ERC a bit more compact, and have only `exchangeRate` (renamed to `rate`). Before maturity, it returns the rate that will work at maturity, after maturity, it can return the same, or change._
+
+OPTIONAL - This method can be used to provide needed data for the calculation of interest post-maturity, but interfaces and other contracts MUST NOT expect this method to be present. _Alberto: Optionality in ERCs is better avoided. Not everyone will implement it, therefore integrators must assume it might not be there, and leads to a proliferation of `safe*` methods._
+
+MUST _NOT_ revert.
+
+```yaml
+- name: exchangeRate
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: rate
+      type: uint256
+```
+
+#### mint
+
+_Alberto: do we need a `mint` method? I reckon it might be nice for integrators to have a standard function to obtain the PTs, but so far we have been fine without one for ERC20s._
```

Comment:
@alcueca Yeah I am unsure if a `mint` is necessary.

For yield tokenization protocols, `mint` is generally on a central router contract, given when minting you are also minting Yield Tokens as well, while in fixed-yield money-markets, a central contract handles usually bookkeeping and thus minting as well.

That said, I noticed its implementation in a couple codebases (including Yield's) so included it since redundant minting capabilities through the PT could be marginally useful for Yield Tokenization protocols and it seemed you guys had already included it. Was there any specific rationale towards having `mint` in the fyTokens?

Reply:
We included it in the fyToken as a workaround for some issues with our liquidity rolling contracts. Basically they mint fyToken out of udnerlying, and it was a lot simpler to do it that way, while still correct. It is pretty much an edge case that shouldn't be representative of anything intrisic to fixed yield tokens.